go/vt/vttablet/tabletserver/query_executor_test.go

```diff
@@ -1261,165 +1262,137 @@ func TestReplaceSchemaName(t *testing.T) {
 		},
-		queries: []string{
-			"select * from t limit 10001",
-			// The previous query isn't passed to the query consolidator,
-			// so the next query can't consolidate into it.
-			"select * from t limit 10001",
-			"select * from t limit 10001",
-			// This query should consolidate into the previous query
-			// that was passed to the consolidator.
-			"select * from t limit 10001",
+		{
+			consolidatorEnabledByDefault:  true,
+			consolidatorExecuteOption:     querypb.ExecuteOptions_CONSOLIDATOR_DISABLED,
+			consolidatorHasIdenticalQuery: true,
+			expectConsolidate:             false,
+			expectExec:                    true,
+			input:                         "select * from t limit 10001",
 		},
-	}, {
-		consolidates: []bool{
-			false,
-			true,
-			false,
-			true,
-			false,
+		{
+			consolidatorEnabledByDefault:  false,
+			consolidatorExecuteOption:     querypb.ExecuteOptions_CONSOLIDATOR_DISABLED,
+			consolidatorHasIdenticalQuery: true,
+			expectConsolidate:             false,
+			expectExec:                    true,
+			input:                         "select * from t limit 10001",
 		},
-		executorFlags: enableConsolidator,
-		name:          "consolidator=enabled",
-		options: []querypb.ExecuteOptions_Consolidator{
-			querypb.ExecuteOptions_CONSOLIDATOR_UNSPECIFIED,
-			querypb.ExecuteOptions_CONSOLIDATOR_UNSPECIFIED,
-			querypb.ExecuteOptions_CONSOLIDATOR_DISABLED,
-			querypb.ExecuteOptions_CONSOLIDATOR_UNSPECIFIED,
-			querypb.ExecuteOptions_CONSOLIDATOR_DISABLED,
+		{
+			consolidatorEnabledByDefault:  false,
+			consolidatorExecuteOption:     querypb.ExecuteOptions_CONSOLIDATOR_ENABLED,
+			consolidatorHasIdenticalQuery: false,
+			expectConsolidate:             true,
+			expectExec:                    true,
+			input:                         "select * from t limit 10001",
 		},
-		queries: []string{
-			"select * from t limit 10001",
-			"select * from t limit 10001",
-			// This query shouldn't be passed to the consolidator.
-			"select * from t limit 10001",
-			"select * from t limit 10001",
-			// This query shouldn't be passed to the consolidator.
-			"select * from t limit 10001",
+		{
+			consolidatorEnabledByDefault:  false,
+			consolidatorExecuteOption:     querypb.ExecuteOptions_CONSOLIDATOR_ENABLED,
+			consolidatorHasIdenticalQuery: true,
+			expectConsolidate:             true,
+			expectExec:                    false,
+			input:                         "select * from t limit 10001",
 		},
-	}}
-	for _, tcase := range testcases {
-		t.Run(tcase.name, func(t *testing.T) {
+	}
+	for _, tcase := range testCases {
+		name := fmt.Sprintf("table-consolidator:%t;query-consolidator:%v;identical-query:%t",
+			tcase.consolidatorEnabledByDefault, tcase.consolidatorExecuteOption, tcase.consolidatorHasIdenticalQuery)
+		t.Run(name, func(t *testing.T) {
+			// Set up fake db, tablet server (with fake consolidator), and executor.
+
 			db := setUpQueryExecutorTest(t)
```

Comment:
Curious if we really need to create this, the tablet server, and query executor for every test case? It's fine if we do.

Reply:
Might not need to, let me see how hard that is to change